### DISCUSSION TITLE
Chakra-UI snippets typescript problem, especially with children prop

### DISCUSSION THREAD

**Discussion by godsamit** [chakra-ui/chakra-ui#9306] [Q&A] [answered]
I'm trying to migrate to V3. Some snippets don't fit my project's lint configurations, so I had to go in and fix one by one.

Now I'm encountering this issue:
![image](https://github.com/user-attachments/assets/25b4650d-b924-4af3-b091-03f2ca5a6f10)

This happens in all the snippets with children. How do I fix the type? 

BTW I could add the children definition in the snippet but still doesn't fix it as the underlying ark ui is expecting a different type I guess:
![image](https://github.com/user-attachments/assets/e0be6df4-28e8-4611-bf26-766592d0a84a)

**Comment by segunadebayo** [comment]
Can you share your TSConfig setup?

Please check that the `target` and `module` are both set to `ESNext`

**Reply by zhbdripon** [reply]
I am facing the same issue and I can confirm target and module in both of my `tsconfig.json` and `tsconfig.node.json`  are set as you suggested

`{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "allowJs": false,
    "skipLibCheck": true,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"],
    }
  },
  "include": ["src"],
  "references": [{ "path": "./tsconfig.node.json" }],
  "types": [ "node"],
}
`

`  "compilerOptions": {
    "composite": true,
    "target": "ESNext",
    "module": "ESNext",
    "moduleResolution": "Node",
    "allowSyntheticDefaultImports": true,
  },
  "include": ["vite.config.ts"],`

**Reply by rexfordessilfie** [reply]
I can confirm that this does not work for me as well. Apparently React 18 no longer has automatic `children` prop for components. Could this be related?

https://stackoverflow.com/a/59106817/13944042

**Reply by rexfordessilfie** [reply]
Found a solution that works for me. You have to set `moduleResolution: "bundler"`.

**Comment by david-morris** [comment]
I don't think Chakra-UI snippets even found my typescript config.  I'm using NX and didn't find any options for specifying a path of the typescript config; this is needed since it needs to find `typescript.lib.json` in a subdirectory.

**Comment by segunadebayo** [comment, marked as answer]
Duplicate of https://github.com/chakra-ui/chakra-ui/discussions/9038

Same snippet TypeScript error on `children`. Align `@types/react` with your React version, or regenerate snippets on a current CLI.
